trieste/acquisition/rule.py

```diff
@@ -1055,10 +1197,38 @@ def acquire(
             points from the previous acquisition state.
         """
 
+        # Subspaces should be set by the time we call `acquire`.
+        assert self._tags is not None
+        assert self._init_subspaces is not None
+
+        num_local_models: Dict[Tag, int] = defaultdict(int)
+        for tag in models:
+            ltag = LocalizedTag.from_tag(tag)
+            if ltag.is_local:
+                num_local_models[ltag.global_tag] += 1
+        num_local_models_vals = set(num_local_models.values())
+        assert (
+            len(num_local_models_vals) <= 1
+        ), f"The number of local models should be the same for all tags, got {num_local_models}"
+        _num_local_models = 0 if len(num_local_models_vals) == 0 else num_local_models_vals.pop()
+
+        num_subspaces = len(self._tags)
+        assert _num_local_models in [0, num_subspaces], (
+            f"When using local models, the number of subspaces {num_subspaces} should be equal to "
+            f"the number of local models {_num_local_models}"
+        )
+
+        # If we have local models, run the (deepcopied) base rule sequentially for each subspace.
+        # Otherwise, run the base rule as is, once with all models and datasets.
+        # Note: this should only trigger on the first call to `acquire`, as after that we will
```

Comment:
I'm confused as to what prevents this triggering on subsequent calls to acquire? Did you mean to include `and self._rules is not None`?

Reply:
Good spot. Yes an expression is missing. It got lost in translation from a previous implementation.